packages/sanes-chrome-extension/src/logic/user/accountmanager.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Algorithm,
+  ChainId,
+  publicIdentityEquals,
+  PublicIdentity,
+} from '@iov/bcp';
+import { UserProfile, WalletId } from '@iov/core';
+import { Slip10RawIndex } from '@iov/crypto';
+import { ReadonlyWallet } from '@iov/keycontrol/types/wallet';
+
+export interface AccountInfo {
+  readonly name: string;
+  readonly identities: ReadonlyMap<ChainId, PublicIdentity>;
+}
+
+export interface AccountManagerChainConfig {
```

Comment:
This is the part I would give another spin. Correct me if my assumptions are wrong.

You are trying to reduce complexity in Persona, exporting all the accounts related internal methods outside. Placing it inside the user folder. At this point, I see two weaknesses:
- User fights with persona and can mislead.
- mnemonic method is here because it needs the user profile, it is not related semantically with Account.

I do like extracting logic for making simpler the persona, but I suggest change folder into something like:
```
logic/index.ts // methods to be used by the app for recreating persona, also for having the method if it is locked?
logic/persona/persona.ts // persona entity
logic/persona/personaFactory.ts //static class used to keep identity in a singleton way having functionality to clear memory if needed, being ready to grow once we support multiple personas.
logic/persona/accountGenerator.ts // file containing logic for generating accounts. Keep in mind in the future this probably will change, in order to generate names in a deterministic way. And placing in one single file all "dense" methods we have created is great because it does not break the open-close principle.
```

Having this file as it is, once we need to change something on how we generate accounts we are going to modify modules which contain an absolutely not related logic.

Reply:
Thanks for your feedback. There are some easy improvements I can make directly based on this:
- Move accountmanager.ts into the logic/persona folder since it is a direct dependency of the Persona. Not export `AccountManager` in `logic/persona/index.ts` because it should only be used inside the folder, not from outside.
- Remove `AccountManager.mnemonic`, which can go to the Persona entirely